packages/target-decisioning-engine/src/geoProvider.js

```diff
@@ -0,0 +1,79 @@
+import {
+  getFetchApi,
+  isDefined,
+  isUndefined,
+  isValidIpAddress
+} from "@adobe/target-tools";
+
+import { getGeoLookupPath } from "./utils";
+
+/**
+ * @param { Response } fetchResponse;
+ * @return { import("@adobe/target-tools/delivery-api-client/models/Geo").Geo } geoObject
+ */
+export function createGeoObject(fetchResponse) {
+  const geoObject = {};
+
+  const ipAddress = fetchResponse.headers.get("X-Forwarded-For");
+  if (ipAddress != null && isDefined(ipAddress)) {
+    geoObject.ipAddress = ipAddress;
+  }
+  const latitude = fetchResponse.headers.get("X-GEO-Latitude");
+  if (latitude != null && isDefined(latitude)) {
+    geoObject.latitude = parseFloat(latitude);
+  }
+  const longitude = fetchResponse.headers.get("X-GEO-Longitude");
+  if (longitude != null && isDefined(longitude)) {
+    geoObject.longitude = parseFloat(longitude);
+  }
+  const countryCode = fetchResponse.headers.get("X-GEO-Country-Code");
+  if (countryCode != null && isDefined(countryCode)) {
+    geoObject.countryCode = countryCode;
+  }
+  const stateCode = fetchResponse.headers.get("X-GEO-Region-Code");
+  if (stateCode != null && isDefined(stateCode)) {
+    geoObject.stateCode = stateCode;
+  }
+  const city = fetchResponse.headers.get("X-GEO-City");
+  if (city != null && isDefined(city)) {
+    geoObject.city = city;
+  }
+  return geoObject;
+}
+
+/**
+ * @param {import("../types/DecisioningConfig").DecisioningConfig} config
+ * @param { import("../types/DecisioningArtifact").DecisioningArtifact } artifact
+ */
+export function GeoProvider(config, artifact) {
+  const fetchApi = getFetchApi(config.fetchApi);
+  const { geoTargetingEnabled = false } = artifact;
+
+  /**
+   * @param {import("@adobe/target-tools/delivery-api-client/models/Geo").Geo} geoRequestContext
+   * @return { import("@adobe/target-tools/delivery-api-client/models/Geo").Geo } geoObject
```

Comment:
The return for jsdoc should not have an object name.

Reply:
Removed